src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -0,0 +1,144 @@
+(ns puppetlabs.services.legacy-routes.legacy-routes-core
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [ring.util.codec :as ring-codec]
+            [compojure.route :as route]
+            [clojure.tools.logging :as log]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; v3 => v4 header translation support functions
+
+; TODO: Improve the naive parsing here with a real library like
+; https://github.com/ToBeReplaced/http-accept-headers
```

Comment:
Seems odd that nothing else would be available in any of the Ring-y middleware libraries that we're already pulling in.  I did find a `parse-accepts-header` in liberator - https://github.com/clojure-liberator/liberator/blob/0.12.0/src/liberator/conneg.clj#L180.  Didn't immediately see a corresponding "flatten" function for that in the namespace, although it wouldn't seem too difficult to write that part, I guess.